blocks/targeted-block/condition-matcher.js

```diff
@@ -12,29 +12,41 @@ export default function conditionsMatched(activeRules, blockConfig) {
     'catalog-price-rules': catalogPriceRules,
   } = blockConfig;
 
-  const activeSegments = activeRules.customerSegments?.map(
-    (segment) => segment.name.trim(),
+  const currentUrlParams = new URLSearchParams(window.location.search);
+  const utmParamsConfig = Object.keys(blockConfig).filter(
+    (key) => (key.indexOf('utm-') === 0),
+  ).reduce(
+    (data, key) => {
+      data[key.replace('-', '_')] = blockConfig[key];
+      return data;
+    },
+    {},
   );
-  const activeGroup = activeRules.customerGroup?.name.trim();
-  const activeCartRules = activeRules.cart?.map(
-    (rule) => rule.name.trim(),
-  );
-  const activePriceRules = activeRules.catalogPriceRules?.rules?.map(
-    (rule) => rule.name.trim(),
+
+  const nonMatchingUtmParams = Object.keys(utmParamsConfig).filter(
+    (key) => (currentUrlParams.get(key.replace('-', '_')) !== utmParamsConfig[key]),
```

Comment:
question: can't we just use `_` instead of `-` in targeted block's config to avoid those replacements?

Reply:
`_` in the block configuration is replaced with `-`, so we have to replace them back